olive/systems/docker/docker_system.py

```diff
@@ -149,43 +150,32 @@ def _run_container(self, tempdir, model: OliveModel, metrics: List[Metric], cont
 
         run_command = docker_utils.create_run_command(run_params=self.run_params)
 
-        try:
-            logger.debug(f"Running container with eval command: {eval_command}")
-            logger.debug(f"The volumes list is {volumes_list}")
-            container = self.docker_client.containers.run(
-                image=self.image, command=eval_command, volumes=volumes_list, detach=True, **run_command
+        environment = run_command.pop("environment", {})
+        envs_dict = {"PYTHONPYCACHEPREFIX": "/tmp"}
+        for k, v in envs_dict.items():
+            if isinstance(environment, list):
+                environment = {env.split("=")[0]: env.split("=")[1] for env in environment}
+            elif isinstance(environment, dict) and not environment.get(k):
+                environment[k] = v
+
+        logger.debug(f"Running container with eval command: {eval_command}")
+        container = self.docker_client.containers.run(
+            image=self.image,
+            command=eval_command,
+            volumes=volumes_list,
+            detach=True,
+            environment=environment,
+            **run_command,
+        )
+        for line in container.logs(stream=True):
+            logger.info(line.strip().decode())
+        exit_code = container.wait()["StatusCode"]
+        container.remove()
```

Comment:
Please help double check if we can remove the container once the evaluate done.

I comment this out in last PR as I was thinking if we should keep the container for debug. But just found too many containers were created and it is hard to start as the mount path in host may be changed.
![image](https://github.com/microsoft/Olive/assets/13343117/f443e881-6725-43f3-8f21-04ab73f893a0)

Reply:
I think removing the container is a good idea. I found that I had a lot of stopped containers on my machine when trying to remove images.